lib/collection/response.js

```diff
@@ -107,6 +107,8 @@ _.inherit((
 
 _.assign(Response.prototype, /** @lends Response.prototype */ {
     update: function (options) {
+        // options.stream accepts new Buffer() as well as new Buffer().toJSON()
+        options.stream && options.stream.type === 'Buffer' && (options.stream = new Buffer(options.stream));
```

Comment:
@czardoz This is the time, shall we stop assigning body to stream and follow, stream to stream and body to body props.